wordpress-mobile/WordPress-Android
# Jetpack performance settings

This PR adds Jetpack performance settings based on the desings. There is now a `Site accelerator` preference which opens a screen with `Faster images` and `Faster static files` and a combined `Site accelerator` switch which turns both of them on/off. 

On the main screen you see now `Site accelerator`, `Lazy load images`, `Ad-free Video Hosting` and `More`. Clicking on `More` opens a screen where there are all the mentioned options + `Jetpack search`. 

`Ad-free video hosting` is only available for Jetpack sites with a Premium or Business plan.

`Jetpack search` is available to Jetpack sites with a business plan or WP.com sites that have the flag `isJetpackSearchSupported` set to `true`. 

The easiest way to test this is comparing the options in Calypso with the same site to the options in the WPAndroid app. It happens that the Jetpack refuses to set the settings but it happens on both platforms.

To test - with a Jetpack site with a free plan:
* Go to Site settings
* Scroll down
* Under the `Performance` category you see:
 * `Site accelerator`
 * `Lazy load images`
 * `More` button

To test - with a Jetpack site with a premium plan:
* Go to Site settings
* Scroll down
* Under the `Performance` category you see:
 * `Site accelerator`
 * `Lazy load images`
 * `Ad-free Video hosting` option
 * `More` button
* Click on the `More` button
* You see the same options as above
* Try setting the new options

To test - with a Jetpack site with a business plan:
* Go to Site settings
* Scroll down
* Under the `Performance` category you see:
 * `Site accelerator`
 * `Lazy load images`
 * `Ad-free Video hosting` option
 * `More` button
* Click on the `More` button
* You see the same options as above + the `Jetpack search` option
* Try setting the new options

To test - with a WPCom site with a business plan:
* Go to Site settings
* Scroll down
* Under the `Performance` category you see:
 * `Site accelerator`
 * `Lazy load images`
 * `More` button
* Click on the `More` button
* You see the same options as above + the `Jetpack search` option
* Try setting the new options


Desings:
![jetpack_peformance_settings_android-1](https://user-images.githubusercontent.com/1079756/64017203-57e16e00-cb29-11e9-99af-00730fee907e.png)

Implementation:
![jetpack_performance_settings](https://user-images.githubusercontent.com/1079756/64017221-5f087c00-cb29-11e9-8f7a-cc2e2fd855f4.gif)

Update release notes:
